src/System.Private.CoreLib/shared/System/Type.Helpers.cs

```diff
@@ -114,12 +114,12 @@ public bool IsVisible
             }
         }
 
-        public virtual Type[] FindInterfaces(TypeFilter filter, object filterCriteria)
+        public virtual Type[] FindInterfaces(TypeFilter filter, object? filterCriteria)
         {
             if (filter == null)
                 throw new ArgumentNullException(nameof(filter));
 
-            Type[] c = GetInterfaces();
+            Type?[] c = GetInterfaces();
             int cnt = 0;
             for (int i = 0; i < c.Length; i++)
```

Comment:
I'm surprised no `!` is necessary on this `c` reference.  Presumably that's because we haven't annotated `Array.Length` yet and thus it's still oblivious?  If GetInterfaces may actually return null, is this a bug and we should be checking for a null here?

Reply:
yeah it's very confusing, it is not nullable for the array itself, its nullable for array elements, so `c.Length` is not giving warning. We need to mark elements nullable becase of row ` c[i] = null;`

Reply:
Ah, yes, I misread the placement of the `?`.